mbuild/formats/charmm_writer.py

```diff
@@ -0,0 +1,2021 @@
+
+        else:
+            self.masses = np.array([atom.mass for atom in self.structure_box_0_FF.atoms]) / self.mass_conversion_factor
+            self.mass_dict = dict([(self.unique_types.index(atom_type) + 1, mass) for atom_type, mass in zip(self.types, self.masses)])
+
+
+        # added an index so the atom types can be converted to numbers as the type name is to long for insertion into
+        # the pdb and psf files
+        self.atom_types_to_index_value_dict = dict(
+            [(self.unique_types[self.unique_types.index(atom_type)], self.unique_types.index(atom_type) ) for atom_type, mass in
+             zip(self.types, self.masses)])
+
+        self.box_0 = Box(lengths=np.array([0.1 * val for val in self.structure_box_0_FF.box[0:3]]),angles=self.structure_box_0_FF.box[3:6])
+        #Divide by conversion factor
+        self.box_0.maxs /= self.sigma_conversion_factor
+
+        #Internally use nm
+        if self.structure_box_1 != None:
+            self.box_1 = Box(lengths=np.array([0.1 * val for val in self.structure_box_1_FF.box[0:3]]),
+                      angles=self.structure_box_1_FF.box[3:6])
+            # Divide by conversion factor
+            self.box_1.maxs /= self.sigma_conversion_factor
+
+        if self.structure_box_1 != None:
+            self.structure_selection = self.structure_box_0_and_1_FF
+        else:
+            self.structure_selection = self.structure_box_0_FF
+
+        # Non-Bonded forces
+        self.epsilons = np.array(
+            [atom.epsilon for atom in self.structure_selection.atoms]) / self.epsilon_conversion_factor
+        self.sigmas = np.array(
+            [atom.sigma for atom in self.structure_selection.atoms]) / self.sigma_conversion_factor
+        self.forcefields = [atom.type + '_' + atom.residue.name for atom in self.structure_selection.atoms]
+        self.Residues = [atom.residue.name for atom in self.structure_selection.atoms]
+        self.epsilon_dict = dict([(self.unique_types.index(atom_type), epsilon)
+                                    for atom_type, epsilon in zip(self.types, self.epsilons)])
+        self.sigma_dict = dict(
+            [(self.unique_types.index(atom_type), sigma) for atom_type, sigma in zip(self.types, self.sigmas)])
+        self.LJ_1_4_dict = dict(
+            [(self.unique_types.index(atom_type), self.combined_1_4_LJ_dict_per_residue[self.Residues])
+             for atom_type, self.Residues in zip(self.types, self.Residues)])
+        self.forcefield_dict = dict([(self.unique_types.index(atom_type), forcefield)
+                                       for atom_type, forcefield in zip(self.types, self.forcefields)])
+
+        # ensure all 1,4-coulombic scaling factors are the same
+        self.coul_1_4_List = []
+        for p in self.combined_1_4_Coul_dict_per_residue.values():
+            self.coul_1_4_List.append(p)
+        for c in range(0, len(self.coul_1_4_List)):
+            if self.coul_1_4_List[c] != self.coul_1_4_List[0]:
```

Comment:
Can instead make a `set` out of `self.coul_1_4_List` and see if there is only 1 element contained.

Reply:
I changed it to be a set function.